### PR TITLE
Fix infinite loop in TCP header read [13721]

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: src/cpp/rtps/transport/TCPTransportInterface.cpp [hunk 1]
```diff
@@ -972,6 +980,11 @@ bool TCPTransportInterface::Receive(
             close_tcp_socket(channel);
             success = false;
         }
+        else if (!channel->connection_status())
+        {
+            logWarning(DEBUG, "Error reading TCP header: channel disconnected while reading.");
```
Comment (juanlofer-eprosima) [inline src/cpp/rtps/transport/TCPTransportInterface.cpp:985]: Should the log category be different?

Comment (JLBuenoLopez-eProsima) [reply src/cpp/rtps/transport/TCPTransportInterface.cpp:985]: Currently there are no too many log levels supported. If you don't want to promote this log to the Error level (even though the message says "Error") then a Warning seems to be correct. If the Warning level is kept, I suggest changing the log message:

```suggestion
            logWarning(DEBUG, "Failed to read TCP header: channel disconnected while reading.");
```

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
I think these changes require a regression test before being considered for merging.

File: src/cpp/rtps/transport/TCPTransportInterface.cpp (hunk 1 above)
Comment (juanlofer-eprosima) [reply src/cpp/rtps/transport/TCPTransportInterface.cpp:985]: Yes I believe it should be a warning, though I was referring to the `DEBUG` part.

Comment (JLBuenoLopez-eProsima) [reply src/cpp/rtps/transport/TCPTransportInterface.cpp:985]: Other Errors/Warnings in this module are being registered under the `DEBUG` label. Thus I will keep consistency. Either change all other logs to another new label (maybe `TCP_DEBUG`) or keep this label as it is currently.

**Comment by juanlofer-eprosima** [comment]
Changes moved to #2470

**Pull request closed without merging**